xapian-core/configure.ac

```diff
@@ -896,12 +918,11 @@ case $enable_backend_glass$enable_backend_honey in
   dnl Check for zlibVersion in -lz.
   SAVE_LIBS=$LIBS
   dnl mingw build needs -lzlib or -lzdll.
-  AC_SEARCH_LIBS([zlibVersion], [z zlib zdll], [], [
+  AC_SEARCH_LIBS([zlibVersion], [z zlib zdll], [
+    XAPIAN_LIBS="$XAPIAN_LIBS -l$ac_lib"
+    ], [
     AC_MSG_ERROR([zlibVersion() not found in -lz, -lzlib, or -lzdll - required for glass (you may need to install the zlib1g-dev or zlib-devel package)])
     ])
-  if test x != x"$LIBS" ; then
-    XAPIAN_LIBS="$XAPIAN_LIBS $LIBS"
-  fi
```

Comment:
`$ac_lib` doesn't seem to be a documented variable.  The `AC_SEARCH_LIBS` docs say:

```
     Search for a library defining FUNCTION if it's not already
     available.  This equates to calling
     'AC_LINK_IFELSE([AC_LANG_CALL([], [FUNCTION])])' first with no
     libraries, then for each library listed in SEARCH-LIBS.

     Prepend '-lLIBRARY' to 'LIBS' for the first library found to
     contain FUNCTION, and run ACTION-IF-FOUND.  If the function is not
     found, run ACTION-IF-NOT-FOUND.
```

Hence the existing code's approach (though I see we don't set `LIBS=` after saving the old value - is that the problem you're addressing with this change?